### PR TITLE
Add config file support.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -13,97 +13,232 @@
...
+		c.args = make(map[string]bool)
+	}
+
+	c.args[flag.Name] = true
 }
 
 func (c *Config) adjust() {
 	if len(c.RootPath) == 0 {
-		c.RootPath = defaultRootPath
+		c.RootPath = "pd"
```
Comment (siddontang) [inline server/config.go]: don't use magic value

Comment (qiuyesuifeng) [inline server/config.go]: But then we should define more and more arguments.
That is terrible.

File: server/config.go [hunk 2]
```diff
@@ -13,97 +13,236 @@
...
+	MetricAddr string `toml:"metric-addr"`
 
 	// Metric prefix.
-	MetricPrefix string
+	MetricPrefix string `toml:"metric-prefix"`
 
-	// For capacity balance.
-	MinCapacityUsedRatio float64
-	MaxCapacityUsedRatio float64
+	BCfg *BalanceConfig `toml:"balance"`
```
Comment (siddontang) [inline server/config.go]: s/BCfg/BalanceConfig/

File: server/config.go [hunk 3]
```diff
@@ -13,97 +13,236 @@
...
 
 	// Only test can change it.
 	nextRetryDelay time.Duration
+
+	args map[string]bool
```
Comment (siddontang) [inline server/config.go]: we don't need to handle flag here, these can be moved to main.

File: server/config.go (hunk 1 above)
Comment (siddontang) [inline server/config.go]: I don't think defining many const vars is a problem.

**Comment by shenli** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**